src/ansible_compat/runtime.py

```diff
@@ -352,6 +352,20 @@ def prepare_environment(
             return
 
         if os.path.exists("galaxy.yml"):
+            if destination:
+                fqcn = fqcn_from_path(os.getcwd())
+                if (
+                    fqcn
+                    and os.path.islink("{destination}/{fqcn}")
+                    and os.path.realpath("{destination}/{fqcn}") != os.getcwd()
+                ):
+                    _logger.warning(
+                        "Collection is symlinked, but not pointing to %s directory. "
+                        "We will remove it.",
+                        os.getcwd(),
+                    )
+                    os.unlink("{destination}/{fqcn}")
```

Comment:
Do we need to raise an error here instead of deleting it?

Reply:
Removing a symlink does not destroy data. We log the action.